packages/query-core/src/queryClient.ts

```diff
@@ -106,11 +107,22 @@ export class QueryClient {
     return this.mutationCache.findAll({ ...filters, fetching: true }).length
   }
 
+  getQueryData<TData = unknown, Filters extends QueryFilters = QueryFilters>(
+    filters: Filters,
+  ): TData | undefined
   getQueryData<TData = unknown>(
     queryKey: QueryKey,
     filters?: QueryFilters,
+  ): TData | undefined
+  getQueryData<TData = unknown, Filters extends QueryFilters = QueryFilters>(
+    arg1: QueryKey | Filters,
+    arg2?: Filters,
   ): TData | undefined {
-    return this.queryCache.find<TData>(queryKey, filters)?.state.data
+    if (isQueryKey(arg1))
+      return this.queryCache.find<TData>(arg1, arg2)?.state.data
+    if (arg1.queryKey)
+      return this.queryCache.find<TData>(arg1.queryKey, arg1)?.state.data
```

Comment:
I think we have a util function that normalizes query key / filters (`parseQueryArgs`)

Reply:
also, this seems like we could pass a filter only without a query key, and that should not be allowed on type level imo.

Reply:
fixed.